### PR TITLE
Update document to Draft 6

### PR DESCRIPTION
**Pull request by mdboom**
This is a work-in-progress branch updating the document to refer to Draft 6.

Ideally, draft 7 will come shortly after Julian/jsonschema supports it, so the examples can continue to be automatically verified.

### CODE REVIEW

**Comment by embray** [comment]
Cool! 👍

File: source/reference/array.rst [hunk 1]
```diff
@@ -34,14 +34,15 @@ array may be of a different type.
...
 
 Items
 '''''
 
 By default, the elements of the array may be anything at all.
 However, it's often useful to validate the items of the array against
-some schema as well.  This is done using the ``items`` and
-``additionalItems`` keywords.
+some schema as well.  This is done using the ``items``,
+``additionalItems`` and ``contains`` keywords.
```
Comment (Relequestual) [inline source/reference/array.rst]: Oxford comma please.

Hey, at least I'm reviewing it!

File: source/reference/generic.rst [hunk 2]
```diff
@@ -8,34 +8,46 @@ for all JSON types.
...
 
 .. _metadata:
 
 Metadata
 --------
 
-JSON Schema includes a few keywords, ``title``, ``description`` and
-``default``, that aren't strictly used for validation, but are used to
-describe parts of a schema.
+JSON Schema includes a few keywords, ``title``, ``description``, ``default`` and
```
Comment (Relequestual) [inline source/reference/generic.rst]: Oxford comma

File: source/reference/numeric.rst [hunk 3]
```diff
@@ -149,46 +149,78 @@ Range
 '''''
 
 Ranges of numbers are specified using a combination of the
-``minimum``, ``maximum``, ``exclusiveMinimum`` and
-``exclusiveMaximum`` keywords.
+``minimum`` and ``maximum`` keywords (or ``exclusiveMinimum`` and
```
Comment (Relequestual) [inline source/reference/numeric.rst]: comma

File: source/reference/schema.rst [hunk 4]
```diff
@@ -21,36 +21,12 @@ Advanced
...
-- ``http://json-schema.org/schema#``
-
-  JSON Schema written against the current version of the
-  specification.
-
-- ``http://json-schema.org/hyper-schema#``
-
-  JSON Schema hyperschema written against the current version of the
-  specification.
+version of the JSON Schema standard, you can include the draft name in the path, for example:
```
Comment (Relequestual) [inline source/reference/schema.rst]: /can/should/

File: source/structuring.rst [hunk 5]
```diff
@@ -139,46 +141,190 @@ schema for a customer:
       }
     }
 
+.. note::
+
+    Even though the value of a ``$ref`` is a URI, it is not a network locator,
```
Comment (Relequestual) [inline source/structuring.rst]: It does not need to be network accessible, but it also may be.

File: source/structuring.rst [hunk 6]
```diff
@@ -139,46 +141,190 @@ schema for a customer:
...
+                "name": "Harry"
+              }
+            ]
+          }
+        ]
+      }
+    }
+
+However, a loop of ``$ref`` schemas referring to one another could cause
+infinite recursion in the validator, and is explicitly disallowed.
```
Comment (Relequestual) [inline source/structuring.rst]: Is it worth expanding this to explain that loops are allowed, but this example shows an unresolveable $ref resolution loop, not a validation loop?

**Comment by mdboom** [comment]
Thanks.  I've updated based on your feedback.  (I'm an Oxford comma fan, but U.S. grad school beat it out of me, so I've ended up rather inconsistent.)

**Comment by handrews** [comment]
@mdboom @Relequestual might be a bit delayed in responding right now, I'll message you on slack with more info

**Comment by mdboom** [comment]
I'm going to merge this to move forward, since I think it's mostly correct.  File issues and PRs for things that aren't!

**Pull request merged**